lib/galaxy/datatypes/sniff.py

```diff
@@ -487,7 +487,7 @@ def handle_uploaded_dataset_file( filename, datatypes_registry, ext='auto', is_m
         ext = guess_ext( filename, sniff_order=datatypes_registry.sniff_order, is_multi_byte=is_multi_byte )
 
     if check_binary( filename ):
-        if not Binary.is_ext_unsniffable(ext) and not datatypes_registry.get_datatype_by_extension( ext ).sniff( filename ):
+        if not datatypes.binary.Binary.is_ext_unsniffable(ext) and not datatypes_registry.get_datatype_by_extension( ext ).sniff( filename ):
```

Comment:
@mvdbeek  I already tried initializing the column_names with out importing Sam class. But column names are not displayed in the galaxy app. Then when I inherited Sam for Bam, colmun names started populating. I donot understand the reason behind.. i think initializing column names alone in Bam class is not enough.